app/src/org/commcare/activities/HomeScreenBaseActivity.java

```diff
@@ -1249,9 +1287,15 @@ public void onResumeSessionSafe() {
             attemptDispatchHomeScreen();
         }
 
+        // In case a Sync was blocked because of a form entry, trigger now if it's safe
+        if (HiddenPreferences.isPostFormSubmissionSyncNeeded() && safeToTriggerBackgroundSyncOnResume) {
```

Comment:
Yes, but end of form navigation could trigger another form entry and in that case `safeToTriggerBackgroundSyncOnResume` is supposed to remain `false`.

Reply:
got it can we add a comment that it's specifically to handle eof navigation, also can rename `safeToTriggerBackgroundSyncOnResume` -> `shouldTriggerBackgroundSync`